components/tidb_query_vec_expr/src/impl_string.rs

```diff
@@ -116,6 +116,37 @@ pub fn rtrim(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
     }))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn lpad(arg: &Option<Bytes>, len: &Option<Int>, pad: &Option<Bytes>) -> Result<Option<Bytes>> {
+    match (arg, len, pad) {
+        (Some(arg), Some(len), Some(pad)) => {
+            if *len < 0 {
+                return Ok(None);
+            }
+            let len = *len as usize;
+            if arg.len() < len && pad.is_empty() {
```

Comment:
Could you add a check on whether target_len larger than the bound `MAX_BLOB_WIDTH`?